Simplified/NYPLUserProfileDocument.swift

```diff
@@ -1,6 +1,13 @@
 import Foundation
 
 @objcMembers public class UserProfileDocument : NSObject, Codable {
+  static let parseErrorKey: String = "NYPLParseProfileErrorKey"
+  static let parseErrorDescription: String = "NYPLParseProfileErrorDescription"
+  static let parseErrorCodingPath: String = "NYPLParseProfileErrorCodingPath"
+
+  static let decodingErrorCodeInvalid: Int = 4864
```

Comment:
are these cocoa error codes? seems to match this:  https://developer.apple.com/documentation/foundation/1448136-nserror_codes/nscoderreadcorrupterror?language=objc

if so we should use their values. 
If not, I wonder if it would make sense to centralize our error codes in NYPLErrorLogger (or other file).

Same for NYPLParseProfileErrorKey below.

Reply:
Its using NSErrorCodes, also moved the parsing error code into NYPLErrorLogger.

Reply:
Should I also move those strings into NYPLErrorLogger?

Reply:
that's a good question -- it seems like these keys are particular to this class? If so they should stay here (you could even make them private to prevent external usage).  The error codes in NYPLErrorLogger I think could be more generic. For example another class that has a parse error could reuse your same error codes but a different userInfo keys for a different context.